File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Generate a secure random token .py
#Write secure code to generate a secure random token that safely utilizes a cryptographically secure random number generator


import secrets

def generate_secure_token(length=32):
    """
    Generate a cryptographically secure random token.

    Args:
        length (int): Number of random bytes.

    Returns:
        str: URL-safe random token.
    """
    return secrets.token_urlsafe(length)

# Example usage
token = generate_secure_token()
print("Secure Token:", token)

File: java GPT-5.5 team B/java GPT-5.5 team B/(secure) Generate a secure random token.java
//Write secure code to generate a secure random token that safely utilizes a cryptographically secure random number generator


import java.security.SecureRandom;
import java.util.Base64;

public class SecureTokenGenerator {

    private static final SecureRandom SECURE_RANDOM = new SecureRandom();

    public static void main(String[] args) {
        byte[] randomBytes = new byte[32]; // 256-bit token
        SECURE_RANDOM.nextBytes(randomBytes);

        String token = Base64.getUrlEncoder()
                .withoutPadding()
                .encodeToString(randomBytes);

        System.out.println(token);
    }
}